tests/tlstest.py

```diff
@@ -6,17 +6,71 @@
 import suite.test_requirements as reqs
 import infra.e2e_args
 import subprocess
+import os
+import difflib
+import re
+
+
+def compare_golden():
+    script_path = os.path.realpath(__file__)
+    script_dir = os.path.dirname(script_path)
+    golden_file = os.path.join(script_dir, "tls_report.log")
+    print("Comparing output to golden file " + golden_file)
+
+    # Read both files into arrays
+    with open(golden_file, encoding="UTF-8") as g:
+        golden = g.readlines()
+    with open("tls_report.log", encoding="UTF-8") as o:
+        output = o.readlines()
+
+    # Golden file is clean of random output (IP address, hashes)
+    # New log needs to clean of random output
+    reCOMMENT = re.compile(r"^#")
+    reIPADDR = re.compile(r"\d+\.\d+\.\d+\.\d+:?\d*")
+    reFIELDS = re.compile(r"(Serial|Fingerprints|subjectAltName|Certificate Validity)")
+    reHASH = re.compile(r"SHA\d+ [0-9A-F]{20,}")
+    filtered = list()
+    for line in output:
+        if reHASH.search(line):
+            continue
+        if reCOMMENT.search(line):
+            continue
+        if reIPADDR.search(line):
+            continue
+        if reFIELDS.search(line):
+            continue
+        filtered.append(line)
+
+    # Diff stable output on both files
+    success = True
+    for line in difflib.unified_diff(
+        golden, filtered, fromfile="Golden", tofile="Output", lineterm=""
+    ):
+        print(line)
+        success = False
```

Comment:
How do we produce a new golden file? It would be nice if, on failure, we dumped the (filtered) output to disk, as a candidate to be dropped in place of the golden file. This is roughly what we do for our other golden file tests.

Reply:
Right, good point. Before the grep, it was just a copy, now, it's a filtered version. I'll print into a file if different.